src/ex6/client/src/components/todo-element.js

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from "react";
+import { deleteItem, markItemAsOld, changeItemStatus } from "../item_client.js";
+import PropTypes from 'prop-types';
+
+export function TodoElement(props) {
+  const [todoClassName, setTodoClassName] = useState("todo-li existing-todo")
+
+  const todo = props.todo;
+  const doneTime = todo.done === null? "" : `Done at ${todo.done.slice(11,16)} ${todo.done.slice(0,10)}`;
+  const checked = todo.status? "checked" : "";
+
+  useEffect(() => {
+    if (todo.isNew) {
+      setTodoClassName("todo-li existing-todo animation-add-todo");
+      markItemAsOld(todo);
+    }
+  },[todo])
+
+  const animationEndHandler = async () => {
+    if (todoClassName === "todo-li existing-todo animation-add-todo") {
+      setTodoClassName("todo-li existing-todo")
+    } else {
+      await props.updateTodos();
+    }
+  }
+
+  const onDeleteButtonClicked = async (e) => {
+    await deleteItem(todo);
+    setTodoClassName("todo-li animation-delete-todo");
+    // updateTodos runs in animationEndHandler
+  }
+
+  const onCheckboxClicked = async (e) => {
+    await changeItemStatus(todo);
+    await props.updateTodos();
+  }
+
+  const todoItem =  <label className="todo-item" info={doneTime}>
+                      <label className="todo-item-checkbox">
+                        <div className="todo-item-text">{todo.text}</div>
+                        <input type="checkbox" className="status-checkbox" defaultChecked={checked} onClick={onCheckboxClicked} />
+                        <span className="status-checkbox-mark"></span>
+                      </label>
+                    </label>
+
+  const deleteButton = <button className="delete-todo-button btn" onClick={onDeleteButtonClicked}>
```

Comment:
extract button into it's own component

Reply:
button.css does not exist anymore. So, this button is just a part of todo-item